flocx_market/models/offer.py

```diff
@@ -0,0 +1,53 @@
+from db import db
+from db import sqlalchemy_jsonfield
+
+class OfferModel(db.Model):
+    __tablename__ = 'offers'
+    marketplace_offer_id = db.Column(db.String(64), primary_key=True, autoincrement=False)
+    provider_id = db.Column(db.String(64), nullable=False)
+    creator_id = db.Column(db.String(64), nullable=False) #username that created the offer
+    marketplace_date_created = db.Column(db.DateTime(timezone=True), nullable=False)
+    status = db.Column(db.String(15), nullable=False, default = 'available')
+    server_id = db.Column(db.String(64), nullable=False, unique=True)
+    start_time = db.Column(db.DateTime(timezone=True), nullable=False)
+    end_time = db.Column(db.DateTime(timezone=True), nullable=False)
+    server_config = db.Column(sqlalchemy_jsonfield.JSONField(enforce_string=True,enforce_unicode=False), nullable=False)
+    cost = db.Column(db.Float, nullable=False)
+
+    def __init__(self, marketplace_offer_id, provider_id, creator_id, marketplace_date_created, status, server_id, start_time, end_time, server_config, cost):
+        self.marketplace_offer_id = marketplace_offer_id
+        self.provider_id = provider_id
+        self.creator_id = creator_id
+        self.marketplace_date_created = marketplace_date_created
+        self.status = status
+        self.server_id = server_id
+        self.start_time = start_time
+        self.end_time = end_time
+        self.server_config = server_config
+        self.cost = cost
+
+    def json(self):
```

Comment:
This method should be renamed. It doesn't return a JSON representation of the data; it returns a dictionary.  So probably something like `as_dict`. You can probably simplify it like this:

```
def as_dict(self):
    return {col.name: getattr(self, col.name)
           for col in self.__table__.columns}
```

This way the method doesn't need to know about specific column names.

Reply:
Agree for the renaming. 
If I try to do this, it throws a bug because the dateTime data types have to be casted to isoformat(). How do you recommend approaching it then?

Reply:
Just maintain a list of datetime columns (e.g. as a class attribute), I guess:

```
def as_dict(self):
    d = {}
    for col in self.__table__.columns:
        if col.name in self.datetime_columns:
            # ...perform conversion here...
        else:
            d[col.name] = getattr(self, col.name)

    return d
```

Reply:
sounds good

Reply:
sounds good